table/block.cc

```diff
@@ -34,23 +34,32 @@ namespace rocksdb {
 // If any errors are detected, returns nullptr.  Otherwise, returns a
 // pointer to the key delta (just past the three decoded values).
 static inline const char* DecodeEntry(const char* p, const char* limit,
-                                      uint32_t* shared,
-                                      uint32_t* non_shared,
-                                      uint32_t* value_length) {
+                                      uint32_t* shared, uint32_t* non_shared,
+                                      uint32_t* value_length,
+                                      const bool value_delta_encoded) {
+  // We need 2 btes for shared and non_shared size. We also need one more byte
+  // either for value size or the actual value in case of value delta encoding.
```

Comment:
Is there a reason we don't always do value delta encoding inside one restart block?

Reply:
Any delta encoding needs one initial full encoding. This implementation uses the first entry in the restart block to have the full encoding. What is the alternative that you would recommend?